core/src/main/java/org/elasticsearch/search/highlight/HighlightBuilder.java

```diff
@@ -164,24 +169,138 @@ public Boolean useExplicitFieldOrder() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("highlight");
+        builder.startObject(HIGHLIGHT_ELEMENT_NAME);
         innerXContent(builder);
         builder.endObject();
         return builder;
     }
 
+    /**
+     * Creates a new {@link HighlightBuilder} from the highlighter held by the {@link QueryParseContext}
+     * in {@link org.elasticsearch.common.xcontent.XContent} format
+     *
+     * @param parseContext
+     *            the input parse context. The state on the parser contained in
+     *            this context will be changed as a side effect of this method
+     *            call
+     * @return the new {@link HighlightBuilder}
+     */
+    public static HighlightBuilder fromXContent(QueryParseContext parseContext) throws IOException {
```

Comment:
I had a look at this already and decided to not do it with this refactoring since the only solutions I found would change the parsing logic quiet a bit and my goal for this was to move over the parsing code from HighlightParseElement with as few modifications as possible. I agree that we should look into this, maybe in a follow-up PR though?